src/topology/metric_space/infsep.lean

```diff
@@ -0,0 +1,464 @@
+    with ⟨_, hxy, hed⟩,
+  simp_rw mem_to_finset at hxy,
+  refine ⟨w.fst, hxy.1, w.snd, hxy.2.1, hxy.2.2, hed⟩
+end
+
+lemma finite.einfsep_exists_of_nontrivial (hsf : s.finite) (hs : s.nontrivial) :
+∃ (x y ∈ s) (hxy : x ≠ y), s.einfsep = edist x y :=
+by { letI := hsf.fintype, exact hs.einfsep_exists_of_finite }
+
+end has_edist
+
+section pseudo_emetric_space
+variables [pseudo_emetric_space α] {x y z : α} {s t : set α}
+
+lemma einfsep_pair (hxy : x ≠ y) : ({x, y} : set α).einfsep = edist x y :=
+begin
+  nth_rewrite 0 [← min_self (edist x y)],
+  convert einfsep_pair_eq_inf hxy using 2,
+  rw edist_comm
+end
+
+lemma einfsep_insert :
+  einfsep (insert x s) = (⨅ (y ∈ s) (hxy : x ≠ y), edist x y) ⊓ (s.einfsep) :=
+begin
+  refine le_antisymm (le_min einfsep_insert_le (einfsep_anti (subset_insert _ _))) _,
+  simp_rw [le_einfsep_iff, inf_le_iff, mem_insert_iff],
+  rintros y (rfl | hy) z (rfl | hz) hyz,
+  { exact false.elim (hyz rfl) },
+  { exact or.inl (infi_le_of_le _ (infi₂_le hz hyz)) },
+  { rw edist_comm, exact or.inl (infi_le_of_le _ (infi₂_le hy hyz.symm)) },
+  { exact or.inr (einfsep_le_edist_of_mem hy hz hyz) }
+end
+
+lemma einfsep_triple (hxy : x ≠ y) (hyz : y ≠ z) (hxz : x ≠ z) :
+  einfsep ({x, y, z} : set α) = edist x y ⊓ edist x z ⊓ edist y z :=
+by simp_rw [einfsep_insert, infi_insert, infi_singleton, einfsep_singleton,
+            inf_top_eq, cinfi_pos hxy, cinfi_pos hyz, cinfi_pos hxz]
+
+lemma einfsep_pi_le_of_le {π : β → Type*} [fintype β] [∀ b, pseudo_emetric_space (π b)]
+  {s : Π (b : β), set (π b)} {c : ℝ≥0∞} (h : ∀ b, c ≤ einfsep (s b) ) :
+  c ≤ einfsep (set.pi univ s) :=
+begin
+  refine le_einfsep (λ x hx y hy hxy, _),
+  rw mem_univ_pi at hx hy,
+  rcases function.ne_iff.mp hxy with ⟨i, hi⟩,
+  exact le_trans (le_einfsep_iff.1 (h i) _ (hx _) _ (hy _) hi) (edist_le_pi_edist _ _ i)
+end
+
+end pseudo_emetric_space
+
+section pseudo_metric_space
+variables [pseudo_metric_space α] {s : set α}
+
+theorem subsingleton_of_einfsep_eq_top (hs : s.einfsep = ∞) : s.subsingleton :=
+begin
+  rw einfsep_top at hs,
+  exact λ _ hx _ hy, of_not_not (λ hxy, edist_ne_top _ _ (hs _ hx _ hy hxy))
+end
+
+theorem einfsep_eq_top_iff : s.einfsep = ∞ ↔ s.subsingleton :=
+⟨subsingleton_of_einfsep_eq_top, subsingleton.einfsep⟩
+
+theorem nontrivial.einfsep_ne_top (hs : s.nontrivial) : s.einfsep ≠ ∞ :=
+by { contrapose! hs, rw not_nontrivial_iff, exact subsingleton_of_einfsep_eq_top hs }
+
+theorem nontrivial.einfsep_lt_top (hs : s.nontrivial) : s.einfsep < ∞ :=
+by { rw lt_top_iff_ne_top, exact hs.einfsep_ne_top }
+
+theorem einfsep_lt_top_iff : s.einfsep < ∞ ↔ s.nontrivial :=
+⟨nontrivial_of_einfsep_lt_top, nontrivial.einfsep_lt_top⟩
+
+theorem einfsep_ne_top_iff : s.einfsep ≠ ∞ ↔ s.nontrivial :=
+⟨nontrivial_of_einfsep_ne_top, nontrivial.einfsep_ne_top⟩
+
+lemma le_einfsep_of_forall_dist_le {d} (h : ∀ (x y ∈ s) (hxy : x ≠ y), d ≤ dist x y) :
+   ennreal.of_real d ≤ s.einfsep :=
```

Comment:
```suggestion
 ennreal.of_real d ≤ s.einfsep :=
```

Reply:
Why does this unindent when the ones indented?

Reply:
It was indented by three spaces instead of two. It looks like my suggestion is doing too much, I just want to bring it back to two spaces.

Reply:
Done.